Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -4,6 +4,170 @@
+                self.enabled = true
+                self.selected = true
+            }
+        }
+    }
+}
+
+class BrowserLocationView : UIView, UIGestureRecognizerDelegate {
+    var delegate: BrowserLocationViewDelegate?
+
+    private var lockImageView: UIImageView!
+    private var locationLabel: UILabel!
+    private var readerModeButton: ReaderModeButton!
+    var readerModeButtonWidthConstraint: NSLayoutConstraint?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.backgroundColor = UIColor.whiteColor()
+        self.clipsToBounds = true
+        self.layer.cornerRadius = 5
+
+        lockImageView = UIImageView(image: UIImage(named: "lock_verified.png"))
+        lockImageView.hidden = false
+        addSubview(lockImageView)
+
+        locationLabel = UILabel()
+        locationLabel.font = UIFont(name: "HelveticaNeue-Light", size: 14)
+        locationLabel.lineBreakMode = NSLineBreakMode.ByClipping
+        locationLabel.userInteractionEnabled = true
+        addSubview(locationLabel)
+
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: "SELtapLocationLabel:")
+        locationLabel.addGestureRecognizer(tapGestureRecognizer)
+
+        readerModeButton = ReaderModeButton(frame: CGRectZero)
+        readerModeButton.hidden = true
+        readerModeButton.addTarget(self, action: "SELtapReaderModeButton", forControlEvents: UIControlEvents.TouchUpInside)
+        addSubview(readerModeButton)
+
+        makeConstraints()
+    }
+
+    private func makeConstraints() {
+        let container = self
+        let padding = UIEdgeInsetsMake(4, 8, 4, 8)
+
+        lockImageView.snp_remakeConstraints { make in
+            make.centerY.equalTo(container).centerY
+            make.left.equalTo(container.snp_left).with.offset(8)
+            make.width.equalTo(self.lockImageView.intrinsicContentSize().width)
+        }
+
+        locationLabel.snp_remakeConstraints { make in
+            make.centerY.equalTo(container.snp_centerY)
+            if self.url?.scheme == "https" {
+                make.left.equalTo(self.lockImageView.snp_right).with.offset(8)
+            } else {
+                make.left.equalTo(container.snp_left).with.offset(8)
+            }
+            if self.readerModeButton.readerModeState == ReaderModeState.Unavailable {
+                make.right.equalTo(container.snp_right).with.offset(-8)
+            } else {
+                make.right.equalTo(self.readerModeButton.snp_left).with.offset(-4)
+            }
+        }
+
+        readerModeButton.snp_remakeConstraints { make in
+            make.centerY.equalTo(container).centerY
+            make.right.equalTo(container.snp_right).with.offset(-4)
+        }
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func intrinsicContentSize() -> CGSize {
+        return CGSize(width: 200, height: 28)
```

Comment:
This line concerns me a bit. Where did these values come from? What happens on different orientations/devices?